opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -191,33 +202,140 @@ private Optional<ExprType> type(String field) {
+        // ignored
+      }
+    }
+
+    if (parsed == null) {
+      // otherwise, throw an error that no formatters worked
+      throw new IllegalArgumentException(
           String.format(
-              "Construct ExprTimestampValue from \"%s\" failed, unsupported date format.", value),
-          e);
+              "Construct ExprTimestampValue from \"%s\" failed, unsupported date format.", value)
+      );
+    }
+
+    return new ExprTimestampValue(parsed);
+  }
+
+  /**
+   * return the first matching formatter as a time without timezone.
+   *
+   * @param value - time as string
+   * @param dateType - field data type
+   * @return time without timezone
+   */
+  private ExprValue parseTimeString(String value, OpenSearchDateType dateType) {
+    for (DateFormatter formatter : dateType.getAllNamedFormatters()) {
+      try {
+        TemporalAccessor accessor = formatter.parse(value);
+        ZonedDateTime zonedDateTime = DateFormatters.from(accessor);
+        return new ExprTimeValue(
+            zonedDateTime.withZoneSameLocal(ZoneId.of("Z")).toLocalTime());
+      } catch (IllegalArgumentException  ignored) {
+        // nothing to do, try another format
+      }
+    }
+
+    // if no named formatters are available, use the default
+    if (dateType.getAllNamedFormatters().size() == 0) {
+      try {
+        return new ExprTimeValue(
+            DateFormatters.from(STRICT_HOUR_MINUTE_SECOND_FORMATTER.parse(value)).toLocalTime());
+      } catch (DateTimeParseException e) {
+        // ignored
+      }
     }
+    throw new IllegalArgumentException("Construct ExprTimeValue from \"" + value
+        + "\" failed, unsupported time format.");
   }
 
-  private ExprValue parseTimestamp(Content value) {
+  /**
+   * return the first matching formatter as a date without timezone.
+   *
+   * @param value - date as string
+   * @param dateType - field data type
+   * @return date without timezone
+   */
+  private ExprValue parseDateString(String value, OpenSearchDateType dateType) {
+    for (DateFormatter formatter : dateType.getAllNamedFormatters()) {
+      try {
+        TemporalAccessor accessor = formatter.parse(value);
+        ZonedDateTime zonedDateTime = DateFormatters.from(accessor);
+        // return the first matching formatter as a date without timezone
+        return new ExprDateValue(
+            zonedDateTime.withZoneSameLocal(ZoneId.of("Z")).toLocalDate());
+      } catch (IllegalArgumentException  ignored) {
+        // nothing to do, try another format
+      }
+    }
+
+    // if no named formatters are available, use the default
+    if (dateType.getAllNamedFormatters().size() == 0) {
+      try {
+        return new ExprDateValue(
+            DateFormatters.from(STRICT_YEAR_MONTH_DAY_FORMATTER.parse(value)).toLocalDate());
+      } catch (DateTimeParseException e) {
+        // ignored
+      }
+    }
+    throw new IllegalArgumentException("Construct ExprDateValue from \"" + value
```

Comment:
The same for exception

Reply:
Discussion to be had for a follow up PR including user defined formats